go/parquet/file/file_reader_test.go

```diff
@@ -271,6 +286,31 @@ func (p *PageSerdeSuite) TestCompression() {
 	}
 }
 
+func TestWithEOFReader(t *testing.T) {
+	sink := encoding.NewBufferWriter(0, memory.DefaultAllocator)
+	serializer := thrift.NewThriftSerializer()
+	fields := schema.FieldList{
+		schema.NewInt32Node("int_col", parquet.Repetitions.Required, -1),
+	}
+	root, _ := schema.NewGroupNode("schema", parquet.Repetitions.Repeated, fields, -1)
+	schema := schema.NewSchema(root)
+	props := parquet.NewWriterProperties(parquet.WithVersion(parquet.V2_LATEST))
+	builder := metadata.NewFileMetadataBuilder(schema, props, nil)
+	fileMetaData, _ := builder.Finish()
+
+	size, _ := serializer.Serialize(fileMetaData, sink, nil)
+
+	binary.Write(sink, binary.LittleEndian, uint32(size))
+	magic := []byte("PAR1")
+	sink.Write(magic)
+	buf := sink.Finish()
+	defer buf.Release()
```

Comment:
You should be able to easily do something like:

```go
root, _ := schema.NewGroupNode("schema", parquet.Repetitions.Repeated, schema.FieldList{
        schema.NewInt32Node("int_col", parquet.Repetitions.Required, -1)}, -1)
props := parquet.NewWriterProperties(parquet.WithVersion(parquet.V2_LATEST)

var sink bytes.Buffer
wr := file.NewParquetWriter(&sink, root, file.WithWriterProps(props))
require.NoError(t, wr.Close())

r := bytes.NewReader(buf.Bytes())
_, err := file.NewParquetReader(testReader{Reader: r})
assert.NoError(t, err)
```
And just make sure that without your change the test fails.

Reply:
Appreciate the help! Much cleaner. Done.